src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -152,6 +184,21 @@ public bool Equals(StringSegment other, StringComparison comparisonType)
             return string.Compare(Buffer, Offset, other.Buffer, other.Offset, textLength, comparisonType) == 0;
         }
 
+        // This handles StringSegment.Equals(string, StringSegment, StringComparison) and StringSegment.Equals(StringSegment, string, StringComparison)
+        // vai the implicit type converter
```

Comment:
The implicit operator has been very useful in HttpAbstractions. I can convert all of the constructor and method parameters on my types to StringValues without breaking any callers or adding extra overloads.

StringSegment is easier to use when it can provide one-to-one replacements for existing string methods. Yes I could rewrite those call sites to .Equals(string), but this is quicker and cleaner.